runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/DataUtil.java

```diff
@@ -304,6 +305,27 @@ public long getNumEncodedBytes() {
       }
       return numEncodedBytes;
     }
+
+    @Override
+    public long getCurrNumSerializedBytes() {
```

Comment:
What is the difference between `getNumSerializedBytes` and `getCurrNumSerializedBytes`?  and why do we need this method?

Reply:
[getNumSerializedBytes](https://github.com/apache/incubator-nemo/blob/74d4a86b5c99418922317be41f0655d2dc48be6a/runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/DataUtil.java#L484) should be called after the actual data is taken out of iterator. and it is used to check whether this task is over or not. 

I don't want to make a side effect by modifying getNumSerializedBytes. 
So I created a new method [getCurrNumSerializedBytes](https://github.com/apache/incubator-nemo/blob/74d4a86b5c99418922317be41f0655d2dc48be6a/runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/DataUtil.java#L494) to get serialized bytes before done.